HLTrigger/Muon/plugins/HLTMuonDimuonL3Filter.cc

```diff
@@ -282,12 +287,15 @@ bool HLTMuonDimuonL3Filter::hltFilter(edm::Event& iEvent,
         }  //MTL loop
 
         if (not l1CandTag_.label().empty() and check_l1match) {
+          auto const propagated = prop.extrapolate(*tk);
+          auto const etaForMatch = propagated.isValid() ? propagated.globalPosition().eta() : cand->eta();
+          auto const phiForMatch = propagated.isValid() ? (double)propagated.globalPosition().phi() : cand->phi();
```

Comment:
Because the `phi()` here does not return a simple `double` but a `Geom::Phi<float, Geom::MinusPiToPi>`, which usually just gets implicitly converted when the function is used. But in this case, the compiler requires the types on both sides of the `?` to be of the same type explicitly, hence the conversion. This does not happen for `eta()`.